### DISCUSSION TITLE
Rename `purge.transform` to `purge.preprocess`?

### DISCUSSION THREAD

**Discussion by paulovieira** [tailwindlabs/tailwindcss#6268] [Ideas]
The [`purge.transform`](https://tailwindcss.com/docs/optimizing-for-production#transforming-content) / [`content.transform`](https://github.com/tailwindlabs/tailwindcss/releases/tag/v3.0.0-alpha.1) API is a somewhat new feature (added in #4469, in may/2021). 

However the name might be a bit confusing because in the tailwind world, "transform" will probably mean the "transform utilities": https://tailwindcss.com/docs/transform

Since there will be a major release soon, I would like to use the opportunity to suggest a different name: `purge.preprocess`.

Some reasons to implement this change:
- the most important: no more name clashing with "transform" (will be easier to find with algolia search)
- this feature is not (yet) well known, I guess, so it won't cause much work when migrating from v2 to v3
- naming is hard, but I think `preprocess` is more meaningful than `transform` because that's what the function does: it receives the raw file and should return some processed version (and the actual work by purgecss or whatever it is, will happen afterwards).

"preprocess" is already used in svelte, where it has the exact role as in tailwind (it receives the raw file; the actual work by the svelte compiler is done afterwards): https://svelte.dev/docs#svelte_preprocess

**Comment by MrOxMasTer** [comment]
It is simply impossible to search for errors on transform on github. The transform property always pops out of css
